python-sdk/pachyderm_sdk/api/pps/extension.py

```diff
@@ -148,3 +155,129 @@ def create_secret(
         ).encode()
 
         return super().create_secret(file=file)
+
+    class CreateDatumClient:
+        """Client for creating datums. Note that this class shouldn't be
+        instantiated directly. Instead, use the `create_datum` method to return
+        a new instance.
+
+
+        Methods
+        -------
+        first_batch(input, number) -> List[DatumInfo]:
+            Returns the first batch of datums for the provided input spec.
+        next_batch(number) -> List[DatumInfo]:
+            Returns subsequent batches of datums. `first_batch` must be called first.
+
+        Examples
+        --------
+        >>> from pachyderm_sdk import Client
+        >>> from pachyderm_sdk.api import pps
+        >>>
+        >>> client = Client()
+        >>> cdc = client.pps.create_datum()
+        >>> input = pps.Input(pfs=pps.PFSInput(repo="repo", glob="/*"))
+        >>>
+        >>> datums = cdc.first_batch(input, number=10)
+        >>> for datum in datums:
+        >>>     print(datum)
+        >>>
+        >>> more_datums = cdc.next_batch(number=10)
+        >>> for datum in more_datums:
+        >>>     print(datum)
+        """
+
+        def __init__(
+            self,
+            create_datum: Callable[[Iterable["CreateDatumRequest"]], Iterator["DatumInfo"]],
+        ):
+            self.queue = SimpleQueue()
+            self.response_it = create_datum(self)
+            self.done = False
+
+        def first_batch(self, input: "Input", number: int) -> List["DatumInfo"]:
```

Comment:
I think we can think about renaming these arguments as well. Since `input` is a special work in python, it might be safer to call this `input_spec`. `number` is a bit ambiguous, maybe `batch_size`?